x/staking/keeper/val_state_change.go

```diff
@@ -184,7 +179,6 @@ func (k Keeper) ApplyAndReturnValidatorSetUpdates(ctx sdk.Context) (updates []ab
 			update := validator.ABCIValidatorUpdate(powerReduction)
 			updates = append(updates, update)
 			// set the validator update and power
-			k.SetValidatorUpdate(ctx, consAddr, update)
```

Comment:
What was this doing and why is it no longer needed?

Reply:
It was setting the validator updates one at a time. I replace it with a single call that sets an array of validator updates. The original code was already collecting this array, i.e., 
```
updates = append(updates, update
```